### PR TITLE
Add null param check and os safe nul pass

### PR DESCRIPTION
**Pull request by amzn-changml**
Signed-off-by: Mike Chang <changml@amazon.com>

## What does this PR do?

- Adds additional checks for null params (as with the case of Multistage pipelines) for non-emgine repo AR runs

- Changes a null pass with an OS compatible command

## How was this PR tested?

Tested in a sandbox branch with o3de and o3de-extras repos

### CODE REVIEW

**Review by byrcolin — APPROVED** [review]

File: scripts/build/Jenkins/Jenkinsfile [hunk 1]
```diff
@@ -624,10 +624,15 @@ def PreBuildCommonSteps(Map pipelineConfig, Map params, String snapshot, String
             if(env.PROJECT_OVERRIDE) {
                 external_project = env.PROJECT_OVERRIDE
             }
-            else if(params.CMAKE_LY_PROJECTS) {
+            else if(params && params.CMAKE_LY_PROJECTS) {
                 external_project = params.CMAKE_LY_PROJECTS
             }
-
+            else {
+                echo "No PROJECT_OVERIDE or CMAKE_LY_PROJECTS  defined"
```
Comment (lumberyard-employee-dm) [inline scripts/build/Jenkins/Jenkinsfile:631]: typo: `PROJECT_OVERIDE`
Also there is an extra space before the work "defined"

```suggestion
                echo "No PROJECT_OVERRIDE or CMAKE_LY_PROJECTS defined"
```

File: scripts/build/Jenkins/Jenkinsfile [hunk 2]
```diff
@@ -624,10 +624,15 @@ def PreBuildCommonSteps(Map pipelineConfig, Map params, String snapshot, String
...
+            }
+            
+            echo "external_project = ${external_project }"
```
Comment (lumberyard-employee-dm) [inline scripts/build/Jenkins/Jenkinsfile:634]: Does groovy resolve variables with a space before the ending right curly brace?
```suggestion
            echo "external_project = ${external_project}"
```

Comment (amzn-changml) [reply scripts/build/Jenkins/Jenkinsfile:634]: It shouldn't, but good catch. This ran on o3de since this condition only triggers in a non-engine repo

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by hultonha — APPROVED** [review]

**Comment by hultonha** [comment]
Thanks for fixing this @amzn-changml! Much appreciated 👍 🥳

**Pull request merged**